### PR TITLE
Spotify auth First Pass

### PR DESCRIPTION
**Pull request by realAndrewCohn**

### CODE REVIEW

File: app/services/spotify_wrapper.rb [hunk 1]
```diff
@@ -1,13 +1,40 @@
 class SpotifyWrapper
+  REDIRECT_URI = 'http://localhost:3000/api/callback'.freeze
```
Comment (nikymorg) [inline app/services/spotify_wrapper.rb]: Maybe put this in a yml file which can supportva different uri for each env?

File: app/services/spotify_wrapper.rb [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+
   def self.user_auth_url
```
Comment (nikymorg) [inline app/services/spotify_wrapper.rb]: I would still argue these are instance methods for the SpotifyWrapper.

File: app/services/spotify_wrapper.rb [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+
+    response = RestClient.post(url, body)
+
+    JSON.parse(response)
+  end
+
+  def self.user_data(access_token)
+    user_data = RestClient.get(
+      'https://api.spotify.com/v1/me',
+      Authorization: "Bearer #{access_token}"
```
Comment (nikymorg) [inline app/services/spotify_wrapper.rb]: It doesn't need curlies???

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   # TODO: remove root when frontend is built
   root to: 'api/users#spotify_login'
```
Comment (nikymorg) [inline config/routes.rb]: Not convinced this is the best url for it.

**Comment by realAndrewCohn** [comment]
@nikymorg will integrate changes

File: app/services/spotify_wrapper.rb (hunk 3 above)
Comment (realAndrewCohn) [reply app/services/spotify_wrapper.rb]: surprisingly no

File: config/routes.rb (hunk 4 above)
Comment (realAndrewCohn) [reply config/routes.rb]: spotify_auth? users#auth?

**Comment by realAndrewCohn** [comment]
@nikymorg going to refactor this again and submit a new pr

**Pull request merged**